### DISCUSSION TITLE
Web components testing for legacy browsers

### DISCUSSION THREAD

**Discussion by fernandofranca** [justeattakeaway/pie#710] [Ideas]
On the back of the `DSW-1013 - Implement dialog polyfill` ticket, I created the `DSW-1073 - Add tests for Modal component dialog polyfill`.

Since the dialog polyfill was added in order to improve legacy browser support, especially for Safari 14, it seems fair to test it in that browser.

During refinement, we tried to discuss the feasibility of that and agreed to create a discussion before refining the ticket.

The intent of the discussion is to check if it's technically reasonable, to have tests focused on specific browser versions.

**Comment by siggerzz** [comment]
Thanks for raising this @fernandofranca!

As mentioned in the Playwright docs, the browser version is controlled by the Playwright dependency in the repo. To add to this, Playwright uses a generic WebKit browser and _not_ the branded Safari browser, as noted [here](https://playwright.dev/docs/browsers#webkit)


It looks like Percy have recently introduced a new feature that allows you to specify a browser version, and your test is then executed on hardware maintained by Percy. It looks like this only supports test frameworks that support W3C Protocol (WebDriver being the most popular), with no indication of if/when Playwright will be supported.

https://docs.percy.io/docs/integrate-functional-testing-with-visual-testing#limitations

**Reply by ashleynolan** [reply]
So does this basically mean, it is possible, but only by using WebDriver for these tests?

**Reply by siggerzz** [reply]
Sadly, yes 🙃

**Reply by ashleynolan** [reply]
Ok – can me and you pick up this conversation next week, just to go over the options potentially? I'll put in a meeting now
